app-hw-37/src/components/Burger.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { ReactNode } from 'react';
+import '../styles/menu.css';
+
+type Props = {
+	defaultState?: boolean,
+	click: () => void,
+	children: ReactNode
+}
+type State = { opened: boolean }
+export class Burger extends React.Component<Props, State> {
+	state = { opened: !!this.props.defaultState }
+	handleClick = () => {
+		this.props.click()
+		this.setState({ opened: !this.state.opened })
+	}
+	render() {
+		return (
+			<button className="menu__btn" onClick={this.handleClick}>
+				<input id="menu__toggle" type="checkbox" />
+				{changeSvg(this.state.opened)}
+			</button >)
+	}
+}
+
+const changeSvg = (defaultState: boolean | undefined) => {
+	if (defaultState) {
+		return <svg id="default" width="24" height="24" fill="white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z" /></svg>
```

Comment:
I recommend putting svg to separate file
https://create-react-app.dev/docs/adding-images-fonts-and-files/#adding-svgs